WordPress/Sniffs/WP/DeprecatedParameterValuesSniff.php

```diff
@@ -0,0 +1,196 @@
+					'version' => '2.2.0',
+				),
+				'siteurl' => array(
+					'alt'     => 'echo esc_url( home_url() )',
+					'version' => '2.2.0',
+				),
+				'text_direction' => array(
+					'alt'     => 'is_rtl()',
+					'version' => '2.2.0',
+				),
+			),
+		),
+		'get_bloginfo' => array(
+			1 => array(
+				'home' => array(
+					'alt'     => 'home_url()',
+					'version' => '2.2.0',
+				),
+				'siteurl' => array(
+					'alt'     => 'home_url()',
+					'version' => '2.2.0',
+				),
+				'text_direction' => array(
+					'alt'     => 'is_rtl()',
+					'version' => '2.2.0',
+				),
+			),
+		),
+		'register_setting' => array(
+			1 => array(
+				'misc' => array(
+					'alt'     => 'another settings group',
+					'version' => '3.0.0',
+				),
+				'privacy' => array(
+					'alt'     => 'another settings group',
+					'version' => '3.5.0',
+				),
+			),
+		),
+		'unregister_setting' => array(
+			1 => array(
+				'misc' => array(
+					'alt'     => 'another settings group',
+					'version' => '3.0.0',
+				),
+				'privacy' => array(
+					'alt'     => 'another settings group',
+					'version' => '3.5.0',
+				),
+			),
+		),
+	);
+
+	/**
+	 * Process the parameters of a matched function.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 * @param array  $parameters      Array with information about the parameters.
+	 *
+	 * @return void
+	 */
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
+		$this->get_wp_version_from_cl();
+		$param_count = count( $parameters );
+		foreach ( $this->target_functions[ $matched_content ] as $position => $parameter_args ) {
+
+			// Stop if the position is higher then the total number of parameters.
+			if ( $position > $param_count ) {
+				break;
+			}
+
+			$parameter_position = $this->phpcsFile->findNext(
+				Tokens::$emptyTokens,
+				$parameters[ $position ]['start'],
+				$parameters[ $position ]['end'] + 1,
+				true
+			);
+
+			$matched_parameter = $this->strip_quotes( $this->tokens[ $parameter_position ]['content'] );
+			if ( ! isset( $parameter_args[ $matched_parameter ] ) ) {
+				continue;
+			}
+
+			$message = 'The "%s" parameter has been deprecated since WordPress version %s.';
+			$data    = array(
+				$matched_parameter,
+				$parameter_args[ $matched_parameter ]['version'],
+			);
+
+			if ( ! empty( $parameter_args[ $matched_parameter ]['alt'] ) ) {
+				$message .= ' Use %s instead.';
+				$data[]   = $parameter_args[ $matched_parameter ]['alt'];
+			}
+
+			$is_error = version_compare( $parameter_args[ $matched_parameter ]['version'], $this->minimum_supported_version, '<' );
+			$this->addMessage(
+				$message,
+				$parameter_position,
+				$is_error,
+				$this->string_to_errorcode( $matched_content . 'Found' ),
```

Comment:
My previous remark about the error code not being properly modular still applies: https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/pull/1366#discussion_r192923365

Reply:
Sorry, I did not fully understand what is needed to make it modular. At the moment the error code would `WordPress.WP.DeprecatedParameterValues.register_settingFound`

Reply:
What I mean by this is that the error code as it it, would silence the message for _two_ different errors: `register_setting()` in combination with having either `misc` or `privacy` as the value for the first parameter. This non-modularness gets worse if there would be more deprecated parameters at different positions in the same function.

So making this truly modular, the error code should encompass:
1. the function detected
2. the position/name of the parameter which is problematic
3. the value detected

The current errorcode only takes the first bit into account.

I'm not saying that's the way to go as that may be overdoing it, but either make the errorcode modular or not - i.e. you could just have `Found` as the errorcode -.